lisa/sut_orchestrator/azure/common.py

```diff
@@ -193,25 +206,141 @@ class AzureVmPurchasePlanSchema:
+                default_values=[
+                    schema.DiskControllerType.SCSI,
+                    schema.DiskControllerType.NVME,
+                ],
+            )
+        ),
+    )
+    hyperv_generation: Optional[
+        Union[search_space.SetSpace[int], int]
+    ] = field(  # type:ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            is_allow_set=True,
+            items=[1, 2],
+        ),
+        metadata=field_metadata(
+            decoder=partial(search_space.decode_set_space_by_type, base_type=int)
+        ),
+    )
     network_data_path: Optional[
         Union[search_space.SetSpace[schema.NetworkDataPath], schema.NetworkDataPath]
     ] = field(  # type: ignore
         default_factory=partial(
             search_space.SetSpace,
+            is_allow_set=True,
             items=[
                 schema.NetworkDataPath.Synthetic,
                 schema.NetworkDataPath.Sriov,
             ],
         ),
         metadata=field_metadata(
             decoder=partial(
-                search_space.decode_set_space_by_type, base_type=schema.NetworkDataPath
+                search_space.decode_set_space_by_type,
+                base_type=schema.NetworkDataPath,
             )
         ),
     )
+    security_profile: Union[
+        search_space.SetSpace[SecurityProfileType], SecurityProfileType
+    ] = field(  # type:ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            is_allow_set=True,
+            items=[
+                SecurityProfileType.Standard,
+                SecurityProfileType.SecureBoot,
+                SecurityProfileType.CVM,
+                SecurityProfileType.Stateless,
+            ],
+        ),
+        metadata=field_metadata(
+            decoder=partial(
+                search_space.decode_nullable_set_space,
+                base_type=SecurityProfileType,
+                is_allow_set=True,
+                default_values=[
+                    SecurityProfileType.Standard,
+                    SecurityProfileType.SecureBoot,
+                    SecurityProfileType.CVM,
+                    SecurityProfileType.Stateless,
+                ],
+            )
+        ),
+    )
+
+    def load_features(self, platform: "AzurePlatform") -> None:
+        """
+        Load image features from Azure platform.
+        Relevant image tags will be used to populate the schema.
+        """
+        raw_features = self._get_info(platform)
+        if raw_features:
+            self._parse_info(raw_features)
+
+    def _get_info(self, platform: "AzurePlatform") -> Optional[Dict[str, Any]]:
+        """Get raw image tags from Azure platform."""
+        raise NotImplementedError()
+
+    def _parse_info(self, raw_features: Dict[str, Any]) -> None:
+        """Parse raw image tags to AzureImageSchema"""
+        raise NotImplementedError()
+
+
+def _get_image_features(image: Any) -> Dict[str, Any]:
```

Comment:
it can be called in `load_features` once.

Reply:
I see what you're saying, but I want `_get_info` to return a single type. If I move `_get_image_features` (now called `_get_image_tags` out of `_get_info` then that function would return a MarketplaceImage / GalleryImage / CommunityGalleryImage and it would return None for VHD.

Reply:
You can override `load_from_platform` for vhd. So, the logic applies to other schemas only.

Reply:
That doesn't address the need for `_get_info` to have a common return type. This function takes the Marketplace / SIG / CGI and outputs a common dictionary. The reason I wrote a separate function is to take advantage of the common structure of the three classes.